stdlib/random.mli

```diff
@@ -148,6 +148,25 @@ module State : sig
       correlation.  Both PRNGs can be split later, arbitrarily many times.
       @since 5.0.0 *)
 
+  val to_bytes : t -> Bytes.t
+  (** Serialize the PRNG state to an array of bytes.
+
+      Note that the serialization format may differ across OCaml
+      versions.
+
+      @since 5.1.0
+  *)
+
+  val of_bytes : Bytes.t -> t
+  (** Deserialize an array of bytes representing a PRNG state.
+
+      Note that the serialization format may differ across OCaml
+      versions.
+
+      @raise Invalid_argumnt if the input is in the expected format.
```

Comment:
`Invalid_argmu_e_nt` … is *_not_* in the expected format.

Note that since there is no function to check if the given `Bytes.t` are in the expected format this will lead to people having to catch `Invalid_argument` which they are not supposed to. I'd say either raise `Failure` or provide a validation function.

Reply:
Good point, let's move to `Failure`.